### PR TITLE
[SPARK-32191][FOLLOW-UP][PYTHON][DOCS] Indent the table and reword the main page in migration guide

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is a minor followup to fix:

1. Slightly reword the wording in the main page.

2. The indentation in the table at the migration guide;

    from
  
    Screen Shot 2020-09-01 at 1 53 40 PM

    to

    Screen Shot 2020-09-01 at 1 53 26 PM

### Why are the changes needed?

In order to show the migration guide pretty.

### Does this PR introduce _any_ user-facing change?

Yes, this is a change to user-facing documentation.

### How was this patch tested?

Manually built the documentation.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @viirya, can you take a quick look when you're available?

**Review by viirya — APPROVED** [review]
LGTM, thanks for the fixing.

File: python/docs/source/migration_guide/index.rst [hunk 1]
```diff
@@ -20,11 +20,9 @@
 Migration Guide
 ===============
 
-Migration Guide: PySpark (Python on Spark)
-
-Note that this migration guide describes the items specific to PySpark.
-Many items of SQL migration can be applied when migrating PySpark to higher versions.
-Please refer `Migration Guide: SQL, Datasets and DataFrame <http://spark.apache.org/docs/latest/sql-migration-guide.html>`_.
+This page describes the migration guide specific to PySpark.
+Many items of other migration guides can be applied when migrating PySpark to higher versions because PySpark internally shares other components.
```
Comment (viirya) [inline python/docs/source/migration_guide/index.rst:24]: nit: ... can also be applied ...

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**